logsmanagement/parser.c

```diff
@@ -0,0 +1,1622 @@
+            if(pch) field_size = pch - field;
+
+            float f = 0;
+
+            char ups_resp_time_d[UPS_RESP_TIME_MAX_LEN];
+            snprintf( ups_resp_time_d, 
+                      field_size > (UPS_RESP_TIME_MAX_LEN - 1) ? UPS_RESP_TIME_MAX_LEN : field_size + 1, 
+                      "%.*s", (int)field_size, field);
+
+            if(memchr(field, '.', field_size)){ // nginx time is in seconds with a milliseconds resolution.
+                if(likely(str2float(&f, ups_resp_time_d) == STR2XX_SUCCESS)){ 
+                    log_line_parsed->ups_resp_time = (int) (f * 1.0E6);
+                }
+                else { 
+                    #if ENABLE_PARSE_WEB_LOG_LINE_DEBUG
+                    error("Error while extracting UPS_RESP_TIME from string");
+                    #endif
+                    log_line_parsed->ups_resp_time = 0;
+                    log_line_parsed->parsing_errors++;
+                }
+            }
+            else{ // unlike in the REQ_PROC_TIME case, apache doesn't have an equivalent here
+                #if ENABLE_PARSE_WEB_LOG_LINE_DEBUG
+                error("Error while extracting UPS_RESP_TIME from string");
+                #endif
+                log_line_parsed->ups_resp_time = 0;
+                log_line_parsed->parsing_errors++;
+            }
+            if(verify){
+                if(unlikely(log_line_parsed->ups_resp_time < 0)){
+                    #if ENABLE_PARSE_WEB_LOG_LINE_DEBUG
+                    error("UPS_RESP_TIME is invalid (<0)");
+                    #endif
+                    log_line_parsed->ups_resp_time = 0;
+                    log_line_parsed->parsing_errors++;
+                }
+            }
+            #if ENABLE_PARSE_WEB_LOG_LINE_DEBUG
+            debug(D_LOGS_MANAG, "Extracted UPS_RESP_TIME:%d", log_line_parsed->ups_resp_time);
+            #endif
+
+            goto next_item;
+        }
+
+        if(fields_format[i] == SSL_PROTO){
+            #if ENABLE_PARSE_WEB_LOG_LINE_DEBUG
+            debug(D_LOGS_MANAG, "Item %d (type: SSL_PROTO):%.*s", i, (int)field_size, field);
+            #endif
+
+            if(field[0] ==  '-' && field_size == 1) { 
+                log_line_parsed->ssl_proto[0] = '\0';
+                log_line_parsed->parsing_errors++;
+                goto next_item;
+            }
+
+            #if ENABLE_PARSE_WEB_LOG_LINE_DEBUG
+            debug(D_LOGS_MANAG, "SSL_PROTO field size:%zu", field_size);
+            #endif
+
+            snprintf( log_line_parsed->ssl_proto, 
+                      field_size > (SSL_PROTO_MAX_LEN - 1) ? SSL_PROTO_MAX_LEN : (field_size + 1), 
+                      "%.*s", (int)field_size, field); 
+
+            #if ENABLE_PARSE_WEB_LOG_LINE_DEBUG
+            debug(D_LOGS_MANAG, "log_line_parsed->ssl_proto:%s", log_line_parsed->ssl_proto);
+            #endif
+
+            if(verify){
+                if(unlikely(strcmp(log_line_parsed->ssl_proto, "TLSv1") && 
```

Comment:
I know it is not common to use the `TLS1.0` nowadays, but it exists and when netdata is installed on old distribution it can be detected, so I suggest to add all [SSL/TLS](https://www.wolfssl.com/differences-between-ssl-and-tls-protocol-versions/) versions.

Reply:
In theory `TLSv1` could cover this, but if some developer used `1.0` we are not covering,